ArduCopter/Copter.cpp

```diff
@@ -156,6 +156,8 @@ const AP_Scheduler::Task Copter::scheduler_tasks[] = {
     SCHED_TASK_CLASS(AP_Logger,      &copter.logger,           periodic_tasks, 400, 300),
 #endif
     SCHED_TASK_CLASS(AP_InertialSensor,    &copter.ins,                 periodic,       400,  50),
+    SCHED_TASK_CLASS(Analyse_Noise,         &copter.analyse_noise,         periodic,       400,  50),
+    SCHED_TASK_CLASS(Analyse_Noise,         &copter.analyse_noise,         analyse,        400, 100),
```

Comment:
@andyp1per  the periodic task is being called at 400hz however  you allow users to specify sample frequency up to 1000hz.  How are you able to do this if you are adding samples to your buffer at a rate of 400hz?

Reply:
The sample frequency is actually limited to the loop rate. This is in the unlikely event you want it lower, but I'm certain this parameter is not correct as of yet.

Reply:
I still don’t get it. If you are scheduling it for 400 Hz how can you be getting it at the loop rate of the fast loop. I would think that he would have to be in the fast loop to be then scheduled according to what you specify as your fast loop rate via the parameter.